### PR TITLE
Tech admin cabinet

### PR DESCRIPTION
**Pull request by tiana25**

### CODE REVIEW

File: src/app/header/header.component.html [hunk 1]
```diff
@@ -99,17 +102,23 @@ <h2 class="header_descr">
...
-
-      <a [routerLink]="'./personal-cabinet/' + (user$ | async).role + '/info'"> <button mat-menu-item>Інформація про
-          {{roles[(user$ | async).role]}}</button></a>
-      <a [routerLink]="'./personal-cabinet/workshops'"><button mat-menu-item>Мої гуртки</button></a>
-      <a [routerLink]="'./personal-cabinet/applications'"><button mat-menu-item>Мої заяви</button></a>
-      <!-- <a [routerLink]="'./personal-cabinet/messages'"><button mat-menu-item>Повідомлення</button></a> -->
-      <a *ngIf="(user$ | async).role === Role.parent" [routerLink]="'./personal-cabinet/parent/favorite'"><button
-          mat-menu-item>Улюблене</button></a>
-      <button mat-menu-item (click)="logout()">Вийти</button>
+      <ng-container *ngIf="((user$ | async)?.role !== Role.techAdmin)">
```
Comment (litvinets) [inline src/app/header/header.component.html:105]: I think it is better to do case-switching in this way: 
```suggestion
    <div *ngIf="((user$ | async)?.role !== Role.techAdmin) then UserView; else AdminView"></div>
    <ng-template #UserView></ng-template >
    <ng-template #AdminView></ng-template >
```

In this case, we check the role only once, in your case the role checking if happening twice, for techadmin and for not techadmin

File: src/app/shared/components/sidenav/sidenav.component.html [hunk 2]
```diff
@@ -77,15 +77,24 @@
...
-          <a [routerLink]="'./personal-cabinet/config'" (click)="changeView()">Особиста Інформація</a>
-          <a [routerLink]="'./personal-cabinet/' + (user$ | async).role + '/info'" (click)="changeView()">Інформація про
-            {{roles[(user$ | async).role]}}</a>
-          <a [routerLink]="'./personal-cabinet/workshops'" (click)="changeView()">Мої гуртки</a>
-          <a [routerLink]="'./personal-cabinet/applications'" (click)="changeView()">Мої заяви</a>
-          <!-- <a [routerLink]="'./personal-cabinet/messages'" (click)="changeView()">Повідомлення</a> TODO: add to teh second release-->
-          <a *ngIf="(user$ | async).role === Role.parent" [routerLink]="'./personal-cabinet/parent/favorite'"
-            (click)="changeView()">Улюблене</a>
-          <button (click)="logout()">Вийти</button>
+          <ng-container *ngIf="((user$ | async)?.role !== Role.techAdmin)">
```
Comment (litvinets) [inline src/app/shared/components/sidenav/sidenav.component.html:80]: same as previous comment

File: src/app/shared/store/admin.state.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  @Action(OnUpdateInfoAboutPortalFail)
+  OnUpdateInfoAboutPortalFail({ dispatch }: StateContext<AdminStateModel>, { payload }: OnUpdateInfoAboutPortalFail): void {
+    throwError(payload);
+    dispatch(new ShowMessageBar({ message: 'На жаль виникла помилка', type: 'error' }));
+  }
+
+  @Action(OnUpdateInfoAboutPortalSuccess)
+  OnUpdateInfoAboutPortalSuccess({ dispatch }: StateContext<AdminStateModel>, { payload }: OnUpdateInfoAboutPortalSuccess): void {
+    dispatch(new MarkFormDirty(false));
+    console.log('InfoAboutPortal is updated', payload);
```
Comment (litvinets) [inline src/app/shared/store/admin.state.ts:59]: I believe condole.log we should remove

File: src/app/shell/admin-tools/admin-tools.guard.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  providedIn: 'root'
+})
+export class AdminToolsGuard implements CanLoad {
+  constructor() { }
+
+  @Select(RegistrationState.role)
+  role$: Observable<string>;
+
+  canLoad(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.role$.pipe(filter((role: string) => !!role), map((role: string) => (role === Role.techAdmin)));
```
Comment (litvinets) [inline src/app/shell/admin-tools/admin-tools.guard.ts:19]:
```suggestion
    return this.role$.pipe(filter((role: string) => role === Role.techAdmin), map((role: string) => (role === Role.techAdmin)));
```

**Review by litvinets — CHANGES_REQUESTED** [review]

File: src/app/shared/styles/cabinet.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
+.container {
+  background: #ffffff;
+  box-shadow: 0px 6px 16px rgba(0, 0, 0, 0.08);
+  margin-bottom: 20px;
+  padding: 1.5rem 1.5rem 0 1.5rem;
+  border-radius: 5px;
+
+  .nav {
+  border-bottom: none;
```
Comment (dmitryy90) [inline src/app/shared/styles/cabinet.scss:9]: indentations

File: src/app/shared/styles/cabinet.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  a {
+    font-size: 13px;
+    opacity: 1;
```
Comment (dmitryy90) [inline src/app/shared/styles/cabinet.scss:12]: I cant see where we change opacity. Why do we need set 1 here?

File: src/app/shell/admin-tools/platform/support-edit/support-edit.component.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+   */
+   onDeleteForm(index: number): void {
+    this.SupportFormArray.removeAt(index);
+  }
+
+  onSubmit(): void {
+    if (this.SupportFormArray.invalid) {
+      this.checkValidationAboutFormArray();
+    } else {
+    // this.SupportFormArray.controls.forEach((form: FormGroup) => {});
```
Comment (dmitryy90) [inline src/app/shell/admin-tools/platform/support-edit/support-edit.component.ts:85]: if this code is still needed use TODO: comment

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

**Review by dmitryy90 — APPROVED** [review]

**Review by litvinets — APPROVED** [review]

**Pull request merged**